test/ERC6538Registry.t.sol

```diff
@@ -47,24 +67,296 @@ contract RegisterKeys is ERC6538RegistryTest {
+    bytes memory stealthMetaAddress
+  ) external {
+    (address alice, uint256 alicePk) = makeAddrAndKey(name);
+    SigUtils.RegistrantInfo memory registrantInfo =
+      SigUtils.RegistrantInfo(schemeId, stealthMetaAddress, 0 /* nonce */ );
+    bytes32 hash = sigUtils.getTypedDataHash(registrantInfo);
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
+    bytes memory signature = abi.encodePacked(r, s, v);
+
+    vm.expectEmit(true, true, true, true);
+    emit StealthMetaAddressSet(alice, schemeId, stealthMetaAddress);
+    registry.registerKeysOnBehalf(alice, schemeId, signature, stealthMetaAddress);
+  }
+
+  function testFuzz_Erc1271SignatureIsValid(
+    string memory name,
+    uint256 schemeId,
+    bytes memory stealthMetaAddress
+  ) external {
+    (address alice, uint256 alicePk) = makeAddrAndKey(name);
+
+    vm.prank(alice);
+    ERC1271MockContract erc1271MockContract = new ERC1271MockContract();
+    address registrant = address(erc1271MockContract);
+
+    SigUtils.RegistrantInfo memory registrantInfo =
+      SigUtils.RegistrantInfo(schemeId, stealthMetaAddress, 0 /* nonce */ );
+    bytes32 hash = sigUtils.getTypedDataHash(registrantInfo);
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
+    bytes memory signature = abi.encodePacked(r, s, v);
+
+    vm.expectEmit(true, true, true, true);
+    emit StealthMetaAddressSet(registrant, schemeId, stealthMetaAddress);
+    registry.registerKeysOnBehalf(registrant, schemeId, signature, stealthMetaAddress);
+  }
+
+  function testFuzz_UpdateStealthMetaAddress(
+    string memory name,
+    uint256 schemeId,
+    bytes memory stealthMetaAddress,
+    uint256 numOfUpdates
+  ) external {
+    numOfUpdates = bound(numOfUpdates, 1, 50);
+    (address alice, uint256 alicePk) = makeAddrAndKey(name);
+
+    for (uint256 nonce = 0; nonce < numOfUpdates; nonce++) {
+      SigUtils.RegistrantInfo memory registrantInfo =
+        SigUtils.RegistrantInfo(schemeId, stealthMetaAddress, nonce);
+      bytes32 hash = sigUtils.getTypedDataHash(registrantInfo);
+      (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
+      bytes memory signature = abi.encodePacked(r, s, v);
+
+      vm.expectEmit(true, true, true, true);
+      emit StealthMetaAddressSet(alice, schemeId, stealthMetaAddress);
+      registry.registerKeysOnBehalf(alice, schemeId, signature, stealthMetaAddress);
+    }
+  }
+
+  function testFuzz_RevertIf_Erc712SignatureIsNotValid(
+    string memory name,
+    address bob,
+    uint256 schemeId,
+    bytes memory stealthMetaAddress
+  ) external {
+    ( /*address alice */ , uint256 alicePk) = makeAddrAndKey(name);
+    SigUtils.RegistrantInfo memory registrantInfo =
+      SigUtils.RegistrantInfo(schemeId, stealthMetaAddress, 0 /* nonce */ );
+    bytes32 hash = sigUtils.getTypedDataHash(registrantInfo);
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
+    bytes memory signature = abi.encodePacked(r, s, v);
+
+    vm.expectRevert();
```

Comment:
We can/should expect the specific email here, right?

Reply:
So, I believe the reason we get a revert without any message here and other places is because after failing EIP712 check of`(recoveredAddress == address(0) || recoveredAddress != registrant)`, we go on to check for `IERC1271(registrant).isValidSignature(dataHash, signature)
              != IERC1271.isValidSignature.selector` which is a function call to an EOA address, hence reverting without any error messages and before hitting `revert ERC6538Registry__InvalidSignature()`.
              
For example in `testFuzz_RevertIf_Erc712SignatureIsNotValid`, if we `expectRevert('RandomMessageInsteadOfEmpty')`;

<img width="1280" alt="image" src="https://github.com/ScopeLift/stealth-address-erc-contracts/assets/61768337/989bd208-6295-424a-9da6-17488db779eb">

Reply:
You should be able to do `vm.expectRevert("")` (or maybe you'll need to cast the empty string) to specify that you expect no revert message, whereas `vm.expectRevert()` allows any revert message

Reply:
Added in [fc2a267](https://github.com/ScopeLift/stealth-address-erc-contracts/pull/7/commits/fc2a267daa35427360d802230678342f94ba3bbc)